src/components/button/button.svelte

```diff
@@ -49,7 +49,9 @@
 
   $: tag = href ? 'a' : ('button' as 'a' | 'button')
 
-  const dispatch = createEventDispatcher()
+  const dispatch = createEventDispatcher<{
+    click: CustomEvent
```

Comment:
I think this is actually the type of the `detail` property on the custom event (i.e. `e.detail`, so it should be fine to type this as `{}`
```suggestion
    click: {}
```

Reply:
I updated the code to do it this way, I initially used the `e.detail` property actually, no problem

Just a note about this though: The svelte code makes it seem like it is the `event.detail` property that should end up being passed in the event callback, I noticed in the resulting react components the `CustomEvent` event object is passed at runtime though.

Reply:
Interesting - the type you pass in here is the type of `event.detail`, because the type of the eventArg is `CustomEvent<DetailType>`

https://github.com/sveltejs/svelte/blob/522d06a6add1de1779a87c83bee0a43163044c75/packages/svelte/src/runtime/internal/lifecycle.js#L83

It might be that we're handling this typing wrong with the `gen-react-types` code though :disappointed:

Reply:
After working on this a bit more I think the way this is handled makes sense, there's just some semantics that I initially found confusing:

What I initially expected:
1. Define the event detail type in the `createEventDispatcher<{ event: CustomEvent<DetailType> }>` call in svelte
2. The event handler in react passes `CustomEvent<DetailType>`

How it actually works:
1. Define the event detail type in the `createEventDispatcher<{ event: DetailType }>` call in svelte
2. The event handler in react passes `CustomEvent<DetailType>`

I think this is fine